configuration/src/main/java/io/airlift/configuration/validation/FileExistsValidator.java

```diff
@@ -0,0 +1,57 @@
+package io.airlift.configuration.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static java.lang.String.format;
+
+public class FileExistsValidator
+        implements ConstraintValidator<FileExists, Object>
+{
+    @Override
+    public void initialize(FileExists ignored)
+    {
+        // Annotation has no properties
+    }
+
+    @Override
+    public boolean isValid(Object path, ConstraintValidatorContext context)
+    {
+        if (path == null) {
+            // @NotNull's responsibility
+            return true;
+        }
+
+        boolean fileExists = exists(path);
+
+        if (!fileExists) {
+            context.disableDefaultConstraintViolation();
+            context.buildConstraintViolationWithTemplate(format("file '%s' does not exist", path.toString()))
```

Comment:
It's better to put the variable part at the end:
```java
"File does not exist: " + path
```
`toString()` not required in a format or concatenation context